internal/metrics/metrics.go

```diff
@@ -108,6 +107,65 @@ func (m *recordReadyMetric) Publish() {
 	RecordReady.WithLabelValues(m.name, m.namespace, m.rootHost, strconv.FormatBool(m.delegating)).Set(gauge)
 }
 
+type RemoteClusterCollector struct {
+	Provider *kubeconfigprovider.Provider
+}
+
+func (c *RemoteClusterCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- multiClusterCount
+}
+
+func (c *RemoteClusterCollector) Collect(ch chan<- prometheus.Metric) {
+	ch <- prometheus.MustNewConstMetric(
+		multiClusterCount,
+		prometheus.GaugeValue,
+		float64(len(c.Provider.ListClusters())),
+	)
+}
+
+type LocalCollector struct {
+	Ctx context.Context
+	Mgr manager.Manager
+}
+
+func (c *LocalCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- authoritativeRecordSpecInfo
+}
+
+func (c *LocalCollector) Collect(ch chan<- prometheus.Metric) {
+
+	cl := c.Mgr.GetClient()
+	logger := c.Mgr.GetLogger()
+
+	listOptions := &client.ListOptions{}
+	dnsRecordList := &v1alpha1.DNSRecordList{}
+
+	err := cl.List(c.Ctx, dnsRecordList, listOptions)
+	if err != nil {
+		logger.Info("failed to list dns records", "status", "error")
+	}
+
+	for _, record := range dnsRecordList.Items {
+		if record.IsAuthoritativeRecord() {
+			specString, err := hash.GetCanonicalString(record.Spec)
+			if err != nil {
+				logger.Info("failed to list dns records", "status", "error")
+				continue
+			}
+			value := hash.ToBase36HashLen(specString, 6)
+			ch <- prometheus.MustNewConstMetric(
```

Comment:
the `NewConstMetric` and `Must...` variant seems to be [throw-away](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#bound-service-account-tokens) metric. Does it mean that once the next `Collect()` event happens, all previous metrics are being removed? 
Meaning it will make sure we have metrics only for "live" records

Reply:
Not sure what you are meant to be linking to there. Yes, they are a "throw-away" metric that is not stored in the prometheus cache with in the operator. This means all previous metrics are removed. I have only looked at using this with gauge metric, and am not sure how it would have counter metric. 

Remember it is when we add labels based on the user generated content (resource name), we also need a way to remove of the metric.

Reply:
Oh, the buffer betrayed me. This was the intended link https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#NewConstMetric but you answered it regardless